code/modules/antagonists/clock_cult/scriptures/ocular_warden.dm

```diff
@@ -0,0 +1,67 @@
+#define OCULAR_WARDEN_PLACE_RANGE 4
+#define OCULAR_WARDEN_RANGE 3
+
+/datum/clockcult/scripture/create_structure/ocular_warden
+	name = "Ocular Warden"
+	desc = "An eye turret that will fire upon nearby targets."
+	tip = "Place these around to prevent crew from rushing past your defenses."
+	button_icon_state = "Ocular Warden"
+	power_cost = 500
+	invokation_time = 50
+	invokation_text = list("Summon thee to defend our temple")
+	summoned_structure = /obj/structure/destructible/clockwork/ocular_warden
+	cogs_required = 3
+	category = SPELLTYPE_STRUCTURES
+
+/datum/clockcult/scripture/create_structure/ocular_warden/check_special_requirements()
+	if(!..())
+		return FALSE
+	for(var/obj/structure/destructible/clockwork/structure in get_turf(invoker))
+		to_chat(invoker, "<span class='brass'>You cannot invoke that here, the tile is occupied by [structure].</span>")
+		return FALSE
+	for(var/obj/structure/destructible/clockwork/ocular_warden/AC in range(OCULAR_WARDEN_PLACE_RANGE))
+		to_chat(invoker, "<span class='nezbere'>There is another ocular warden nearby, placing them too close will cause them to fight!</span>")
+		return FALSE
+	return TRUE
+
+/obj/structure/destructible/clockwork/ocular_warden
+	name = "ocular warden"
+	desc = "A wide, open eye that stares intently into your soul. It seems resistant to energy based weapons."
+	clockwork_desc = "A defensive device that will fight any nearby intruders."
+	break_message = "<span class='warning'>A black ooze leaks from the ocular warden as it slowly sinks to the ground.</span>"
+	icon_state = "ocular_warden"
+	max_integrity = 75
+	armor = list("melee" = -80, "bullet" = -50, "laser" = 60, "energy" = 60, "bomb" = 20, "bio" = 0, "rad" = 0)
+	var/cooldown
+
+/obj/structure/destructible/clockwork/ocular_warden/process()
+	//Can we fire?
+	if(world.time < cooldown)
+		return
+	//Check hostiles in range
+	var/list/valid_targets = list()
+	for(var/mob/living/potential in view(OCULAR_WARDEN_RANGE, src))
+		if(!is_servant_of_ratvar(potential) && !potential.stat)
+			valid_targets += potential
+	if(!LAZYLEN(valid_targets))
+		return
+	var/mob/living/target = pick(valid_targets)
+	playsound(get_turf(src), 'sound/machines/clockcult/ocularwarden-target.ogg', 60, TRUE)
+	if(!target)
+		return
+	dir = get_dir(get_turf(src), get_turf(target))
+	target.apply_damage(max(25 - (get_dist(src, target)*5), 10), BURN)
```

Comment:
Pretty sure apply_damage does respect armour.
The laser protection will be lowered to 40, but 10 DPS is actually really low and they die in a couple of hits with their -80 melee protection.

Reply:
Apply damage only respects armor if you set the blocked variable to armor amount iirc, otherwise, it only respects  species armor.

It's also very easy to just set a fuckload of these uo in glass boxes